internal/providers/sdkv2/provider.go

```diff
@@ -1,4 +1,7 @@
-package provider
+// Package sdkv2 contains the changes specific to the SDKv2
```

Comment:
[Nit] try to keep the same name as the package: `sdkv2/provider.go` -> `sdkv2/skv2.go`.